src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1115,6 +1138,8 @@ def get_cls(name, info, parent=None, version=None):
             bases = bases + (_CreatableNamedObjectMixin,)
         elif obj_type == "named-object":
             bases = bases + (_NonCreatableNamedObjectMixin,)
+        elif name in _items_with_string_and_allowed_values:
+            bases = bases + (_StringWithAllowedValuesMixin,)
 
```

Comment:
Do you really require `_filter_items_with_string_and_allowed_values`? Can you not just do the following here:
``` python
if info.get('has-allowed-values'):
  bases += (_HasAllowedValuesMixin,)
```

Reply:
Yes, it is required. If we modify according to above suggestion then following test fails.

![image](https://github.com/ansys/pyfluent/assets/106588300/4e184414-b1b9-47f8-8e7d-b0183a7d69c8)

Reply:
At least in PyConsole mode, this seems to be sufficient:
``` diff
diff --git a/fluent/cortex/pylib/flapi/flobject.py b/fluent/cortex/pylib/flapi/flobject.py
index eb2cb43b9a1..aec98d54045 100644
--- a/fluent/cortex/pylib/flapi/flobject.py
+++ b/fluent/cortex/pylib/flapi/flobject.py
@@ -215,6 +215,8 @@ class Numerical(Property):
 class Textual(Property):
     """Exposes attribute accessor on settings object - specific to string objects."""

+
+class _HasAllowedValuesMixin:
     def allowed_values(self):
         """Get the allowed values of the object."""
         return self.get_attr("allowed-values", (list, str))
@@ -1112,6 +1114,11 @@ def get_cls(name, info, parent=None, version=None):
         elif obj_type == "named-object":
             bases = bases + (_NonCreatableNamedObjectMixin,)

+        if obj_type in ['string'] and 'has-allowed-values' not in info:
+           print ('name =', name, 'info = ', info)
+        if info.get("has-allowed-values", False):
+            bases = bases + (_HasAllowedValuesMixin,)
+
         cls = type(pname, bases, dct)

         taboo = set(dir(cls))

Reply:
![image](https://github.com/ansys/pyfluent/assets/93313177/b7bb8aee-b293-4861-8be4-fc246d1c306f)

Reply:
Done. Corrected above code. Thank you.